stardog-spring/src/main/java/com/complexible/stardog/ext/spring/DataSource.java

```diff
@@ -81,38 +87,39 @@ public Connection getConnection() {
 			throw new RuntimeException(e);
 		}
 	}
-	
+
 	/**
 	 * <code>releaseConnection</code>
 	 * @param connection Stardog Connection
 	 */
-	public void releaseConnection(Connection connection) { 
+	public void releaseConnection(Connection connection) {
 		try {
-			if (pool == null)
-				afterPropertiesSet();
```

Comment:
I took it out because maybe I misunderstood the sequence of events. If the person is trying to remove a connection but there is no pool in which a connection can actually exists why do we re-initialize the bean in order to remove a connection?

Reply:
I think I put that in there for safety, i.e. someone who has manually instantiated the DataSource bean.

The typical flow of events should be the DataSourceFactoryBean creates a DataSource bean, wires the configuration in and then spring will call `afterPropertiesSet` as part of its lifecycle.  So, by the time you get access to a datasource in your SnarlTemplate in your app code, it has been all wired together and initialized.

Anyhow, this line is definitely an odd ball corner defensive case, doesn't hurt anything to leave it in.